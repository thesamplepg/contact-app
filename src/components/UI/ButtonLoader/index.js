import React from "react";
import { faSpinner } from "@fortawesome/free-solid-svg-icons";

import classes from "./index.scss";
import Icon from "../Icon";

const ButtonLoader = () => {
  return (
    <div className={classes.ButtonLoader}>
      <Icon icon={faSpinner} />
    </div>
  );
};

export default ButtonLoader;
