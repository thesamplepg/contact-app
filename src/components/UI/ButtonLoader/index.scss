.ButtonLoader {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.6);

  svg {
    animation: spin 1s linear infinite;
  }

  @keyframes spin {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }
}
