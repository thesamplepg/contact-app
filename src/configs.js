export const inputs = {
  name: {
    type: "text",
    value: "",
    required: 3
  },
  address: {
    type: "text",
    value: "",
    required: 6
  },
  phone: {
    type: "number",
    value: "",
    required: 9
  }
};
