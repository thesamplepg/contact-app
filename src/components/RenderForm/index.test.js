import React from "react";
import { shallow } from "enzyme";

import Component from "./index";
import { inputs } from "../../configs";
import ButtonLoader from "../UI/ButtonLoader";

describe("RenderForm component", () => {
  const props = {
    inputs,
    loading: false,
    errors: []
  };

  const renderComponent = props => shallow(<Component {...props} />);

  it("render propertly", () => {
    expect(renderComponent(props)).toMatchSnapshot();
  });

  it("should render all inputs", () => {
    const component = renderComponent(props);
    expect(component.find("input")).toHaveLength(4);
  });

  it("should render <ButtonLoader/> when loading is true", () => {
    const nextProps = { ...props, loading: true };
    const component = renderComponent(nextProps);
    const buttonWrapper = component.find("button");

    expect(buttonWrapper.contains(<ButtonLoader />)).toBeTruthy();
  });

  it("input[name] should have a class <error>", () => {
    const nextProps = { ...props, errors: ["name"] };
    const component = renderComponent(nextProps);

    const nameInput = component.find('input[id="name"]');

    expect(nameInput.hasClass("error")).toBeTruthy();
  });

  describe("onChange handler", () => {
    it("should call event when user entered smth", () => {
      const changeHandler = jest.fn();

      const nextProps = {
        ...props,
        changeHandler
      };

      const component = renderComponent(nextProps);
      const inputs = component.find("input");

      inputs.forEach(node =>
        node.simulate("change", { target: { value: "asd" } })
      );

      expect(changeHandler).toHaveBeenCalledTimes(3);
    });
  });
});
