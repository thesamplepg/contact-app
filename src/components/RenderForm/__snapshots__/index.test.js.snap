// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderForm component render propertly 1`] = `
<form
  className="AddContactForm"
  encType="multipart/form-data"
  style={
    Object {
      "top": "-100%",
    }
  }
>
  <h2>
    New Contact Address
  </h2>
  <input
    id="avatarInput"
    multiple={false}
    name="avatar"
    type="file"
  />
  <label
    htmlFor="avatarInput"
  >
    Choose Image (Optional)
  </label>
  <input
    className={null}
    id="name"
    key="name"
    placeholder="name"
    type="text"
    value=""
  />
  <input
    className={null}
    id="address"
    key="address"
    placeholder="address"
    type="text"
    value=""
  />
  <input
    className={null}
    id="phone"
    key="phone"
    placeholder="phone"
    type="number"
    value=""
  />
  <p
    className="Error"
  />
  <button
    className="Submit"
    disabled={false}
    onClick={[Function]}
  >
    New Contact
  </button>
</form>
`;
